packages/grafana-ui/src/components/uPlot/config/gradientFills.ts

```diff
@@ -57,35 +58,67 @@ export function getScaleGradientFn(
   }
 
   return (plot: uPlot, seriesIdx: number) => {
+    // A uplot bug (I think) where this is called before there is bbox
+    // Color used for cursor highlight, not sure what to do here as this is called before we have bbox
+    // and only once so same color is used for all points
+    if (plot.bbox.top == null) {
+      return theme.colors.text.primary;
+    }
+
     const ctx = getCanvasContext();
     const gradient = ctx.createLinearGradient(0, plot.bbox.top, 0, plot.bbox.top + plot.bbox.height);
+    const canvasHeight = plot.bbox.height;
     const series = plot.series[seriesIdx];
     const scale = plot.scales[series.scale!];
-    const range = plot.bbox.height;
+    const scaleMin = scale.min ?? 0;
+    const scaleMax = scale.max ?? 100;
+    const scaleRange = scaleMax - scaleMin;
 
-    console.log('scale', scale);
-    console.log('series.min', series.min);
-    console.log('series.max', series.max);
+    const addColorStop = (value: number, color: string) => {
+      const pos = plot.valToPos(value, series.scale!, true);
+      // when above range we get negative values here
+      if (pos < 0) {
+        return;
+      }
 
-    const getColorWithAlpha = (color: string) => {
-      return 'rgb(255, 0, 0)';
-    };
+      const percent = Math.max(pos / canvasHeight, 0);
+      const realColor = tinycolor(theme.visualization.getColorByName(color)).setAlpha(opacity).toString();
```

Comment:
this runs on every data update, and for typical cases (5s refresh) should be ok, but makes me nervous for streaming. would be nice to have a uniform #rgb or #rgba format so we can use the much cheaper colorManipulator#alpha()

https://github.com/grafana/grafana/pull/35911#discussion_r654189622